hat/assets/js/apps/Iaso/components/maps/tools/TilesSwitchControl.tsx

```diff
@@ -0,0 +1,135 @@
+        minHeight: 28,
+        minWidth: 28,
+    },
+    iconContainer: {
+        position: 'absolute',
+        paddingRight: theme.spacing(0.5),
+        paddingTop: theme.spacing(1),
+        top: 0,
+        right: 0,
+        backgroundColor: 'white',
+    },
+    barButton: {
+        display: 'flex',
+        position: 'absolute',
+        backgroundColor: 'white',
+        borderRadius: '4px',
+        padding: '2px',
+        cursor: 'pointer',
+        outline: 'none',
+        top: 0,
+        right: 0,
+        boxShadow: 'none',
+        // @ts-ignore
+        color: `${theme.textColor}! important`,
+    },
+    container: {
+        backgroundColor: 'white',
+        position: 'relative',
+        transition: 'width .1s ease-in-out, height .1s ease-in-out',
+        overflow: 'hidden',
+    },
+    open: {
+        width: 235,
+        height: 240,
+    },
+    closed: {
+        width: 0,
+        height: 0,
+    },
+    title: {
+        paddingLeft: theme.spacing(1),
+        paddingTop: theme.spacing(1),
+    },
+}));
+export type Tile = {
+    maxZoom: number;
+    url: string;
+    attribution?: string;
+};
+type Props = {
+    currentTile: Tile;
+    // eslint-disable-next-line no-unused-vars
+    setCurrentTile: (newTile: Tile) => void;
+};
+
+export const TilesSwitchControl: FunctionComponent<Props> = ({
+    currentTile,
+    setCurrentTile,
+}) => {
+    const { formatMessage } = useSafeIntl();
+    const [tilePopup, setTilePopup] = useState<boolean>(false);
+
+    const classes: Record<string, string> = useStyles();
+
+    const toggleTilePopup = () => {
+        setTilePopup(!tilePopup);
+    };
+
+    return (
+        <>
+            <div className={classes.legendLayers}>
+                {!tilePopup && (
+                    <span
+                        className={classes.barButton}
+                        role="button"
+                        tabIndex={0}
+                        onClick={() => toggleTilePopup()}
+                    >
+                        <Layers fontSize="small" />
+                    </span>
+                )}
+
+                <Box
+                    className={classNames(
+                        tilePopup ? classes.open : classes.closed,
+                        classes.container,
+                    )}
+                >
+                    {tilePopup && (
+                        <Box width={235}>
+                            <Typography
+                                variant="subtitle1"
+                                className={classes.title}
+                            >
+                                {formatMessage(MESSAGES.layersTitle)}
+                            </Typography>
+                            <Box className={classes.iconContainer}>
+                                <IconButton
+                                    size="small"
+                                    onClick={() => toggleTilePopup()}
+                                    icon="clear"
```

Comment:
Not sure about the icon choice here. `Minimize` would  be more consistent, but since it looks like a minus, it would be confusing as well. Not blocking